### ISSUE TITLE
Add optional description field for Contact Block

### ISSUE THREAD

**Issue by ChaosExAnima** [Automattic/jetpack#14188]
#### Is your feature request related to a problem? Please describe.

I would love to have the ability to add a description for fields in the Content Block, to provide additional information for users.

#### Describe the solution you'd like

I believe that having a way in the block to enter a description below the form would be best. An example is as follows:

![Screenshot from 2019-12-05 16-50-59](https://user-images.githubusercontent.com/1760168/70277052-95d0dc80-177f-11ea-9e4b-a3d26f1d5abe.png)

This could allow basic formatting, such as links, bold, etc. We could leverage the `RichText` Core component for this purpose.

#### Describe alternatives you've considered

There's not a good way to currently do this. Right now, the only way to add context for users is to add footnotes essentially, which is not easy to read. Currently most sites I see that need this just embed Google Forms.

#### Additional context

Current FE output:
![Screenshot from 2019-12-05 16-49-20](https://user-images.githubusercontent.com/1760168/70277177-dd576880-177f-11ea-9a49-979cb5edbbab.png)

**Comment by ChaosExAnima** [comment]
@tyxla and @enejb, I believe I mentioned this to you?

**Comment by tyxla** [comment]
Maybe I missed that, but anyway it seems like a pretty useful feature 👍

**Comment by jeherve** [comment]
Wouldn't the placeholder value do the trick there?
![image](https://user-images.githubusercontent.com/426388/70323071-236a0600-182c-11ea-845a-a8213c4baded.png)

![image](https://user-images.githubusercontent.com/426388/70323082-28c75080-182c-11ea-8eec-b9a8c9a21b93.png)

Another alternative is the default value one can set:

![image](https://user-images.githubusercontent.com/426388/70323105-3977c680-182c-11ea-8e65-1d787970a1e3.png)
![image](https://user-images.githubusercontent.com/426388/70323116-40063e00-182c-11ea-9215-33893f37f572.png)

I am a bit weary of adding another field in there as we already have a few, and it may become confusing to add one more.

**Comment by ChaosExAnima** [comment]
@jeherve - The issue is really cases where you need to explain things. For instance, I recently wanted to make a field where users needed to check a box to confirm they read a policy. I can't link the policy right now in the current system. Now, adding a way to add URLs and other formatting would be helpful for the label area, but that falls down if you want something larger, like a paragraph.

I'm open to alternative suggestions, but I don't think placeholders will do it.

**Comment by jeherve** [comment]
ah, that makes sense. Placeholders are also trickier with non-text fields.

> I recently wanted to make a field where users needed to check a box to confirm they read a policy. 

True. This came up in #9623 before, but adding an extra field would be more flexible than just allowing links in one area. 👍

**Comment by ChaosExAnima** [comment]
Yes, I feel there's an elegant way to do this in Gutenberg. We can let users add additional content manually in the shortcode system, as I believe we're not updating that anymore?

**Comment by jeherve** [comment]
> We can let users add additional content manually in the shortcode system, as I believe we're not updating that anymore?

Yeah, I think that would be fine to handle in the block only.
